tensorstore/kvstore/s3/object_metadata_test.cc

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 The TensorStore Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "tensorstore/kvstore/s3/object_metadata.h"
+
+#include <string>
+
+#include <gtest/gtest.h>
+#include "absl/time/time.h"
+#include "tensorstore/kvstore/test_util.h"
+#include "tensorstore/util/result.h"
+#include "tensorstore/util/status.h"
+
+namespace {
+
+using ::tensorstore::internal_storage_s3::ParseObjectMetadata;
+
+
+const char kObjectMetadata[] = R"""({
+  "x-amz-checksum-crc32c": "deadbeef",
+  "x-amz-version-id": "version_12345",
+  "x-amz-delete-marker": false,
+  "Content-Length": "102400",
+  "Last-Modified": "2018-05-19T19:31:14Z"
```

Comment:
These are the S3 metadata headers that seemed relevant to me, based on reading https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingMetadata.html

`x-amz-checksum-sha256` may be the more modern checksum to use. These [docs](https://docs.aws.amazon.com/AmazonS3/latest/userguide/checking-object-integrity.html) suggest is possible to set during upload

> **When you're using the REST API**, you don't use the x-amz-sdk-checksum-algorithm parameter. Instead, you use one of the algorithm-specific headers (for example, x-amz-checksum-crc32).